File: Ruby/season2/p007-009/p008.rb
# -*- coding: utf-8 -*-
=begin
　2017.4.27
　配列操作
=end

def main
  # 配列作成
  array = [1,2,3,4,5,6,7,8,9,"パナソニック","Matsushita_denki",2.71828182846,[5,0,0,0,"兆","円"]];
  puts array
  print array
  p array
  # 配列要素数
  puts "[length]", array.length
  puts "[size]"  , array.size
  # 配列内アクセス
  p array[12]
  puts array[12][0]
  # 代入
  array[12] = "代表なくして課税なし"
  p array
end


main()

File: Python/season2/p007-010/p008.py
# coding: UTF-8
"""
　2018.4.6
　数値 x のシグノイド関数
"""

def sign(x):
  if x > 0:
    return 1
  elif x == 0:
    return 0
  else:
    return -1

a = 1
print("{}\n{}".format(a, sign(a)))
a = -a
print("{}\n{}".format(a, sign(a)))
a = 932894798278727867867834974983267886547386876278346876784368756876278687684730928340
print("{}\n{}".format(a, sign(a)))
a = -52635478712637567325487956782398287643.23847912785236487263790122840939867823674823423798523876427349832794873267832
print("{}\n{}".format(a, sign(a)))
a = 0
print("{}\n{}".format(a, sign(a)))